src/main/java/dev/openfeature/sdk/EventHandling.java

```diff
@@ -0,0 +1,64 @@
+package dev.openfeature.sdk;
+
+import java.util.function.Consumer;
+
+/**
+ * Interface for attaching event handlers.
+ */
+public interface EventHandling<T> {
```

Comment:
Remark - We can also think about a single `addEventHandler()` with `ProviderEvent` enum value as a parameter. But this is just a suggestion and not a requirement.

Reply:
If I'm understanding you correctly, that's the point of "on":

```java
    /**
     * Add a handler for the specified {@link ProviderEvent}.
     * 
     * @param event event type
     * @param handler behavior to add with this event
     * @return this
     */
    T on(ProviderEvent event, Consumer<EventDetails> handler);
```

Reply:
Ohh, I didn't see this clearly. This is done then :)